db/compaction/compaction_job.cc

```diff
@@ -1079,7 +1080,7 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
     CompactionIterationStats range_del_out_stats;
     Status s = FinishCompactionOutputFile(status, sub_compact, &range_del_agg,
                                           &range_del_out_stats);
-    if (status.ok()) {
+    if (!s.ok() && status.ok()) {
```

Comment:
This check `!s.ok()` is unnecessary.

Reply:
The check is necessary to "clear" the checked flag in s.  Otherwise, we may not clear it.